nipype/interfaces/traits_extension.py

```diff
@@ -218,6 +222,69 @@ def __init__(self, value='', auto_set=False, entries=0,
         super(Directory, self).__init__(value, auto_set, entries, exists,
                                         **metadata)
 
+# lists of tuples
+# each element consists of :
+# - uncompressed (tuple[0]) extension
+# - compressed (tuple[1]) extension
+img_fmt_types = {
+        'nifti1': [('.nii', '.nii.gz'),
+                   (('.hdr', '.img'), ('.hdr', '.img.gz'))],
+        'mgh': [('.mgh', '.mgz'), ('.mgh', '.mgh.gz')],
+        'nifti2': [('.nii', '.nii.gz')],
+        'cifti2': [('.nii', '.nii.gz')],
+        'gifti': [('.gii', '.gii.gz')],
+        'dicom': [('.dcm', '.dcm'), ('.IMA', '.IMA'), ('.tar', '.tar.gz')],
+        'nrrd': [('.nrrd', 'nrrd'), ('nhdr', 'nhdr')],
+        'afni': [('.HEAD', '.HEAD'), ('.BRIK', '.BRIK')]
+        }
+
+class ImageFile(File):
+    """ Defines a trait of specific neuroimaging files """
+
+    def __init__(self, value='', filter=None, auto_set=False, entries=0,
+                 exists=False, types=[], allow_compressed=True, **metadata):
+        """ Trait handles neuroimaging files.
+
+        Parameters
+        ----------
+        types : list
+            Strings of file format types accepted
+        compressed : boolean
+            Indicates whether the file format can compressed
+        """
+        self.types = types
+        self.allow_compressed = allow_compressed
+        super(ImageFile, self).__init__(value, filter, auto_set, entries,
+                                        exists, **metadata)
+
+    def grab_exts(self):
+        # TODO: file type validation
```

Comment:
@mgxd : are you still planning to add something more to this validation? or should I remove TODO?